library/datasort.h

```diff
@@ -81,8 +81,10 @@ struct datasort_cfg {
 	struct list_head		sorted_chunks;
 	/* Result of mergesort */
 	struct datasort_chunk		*result;
-	/* Datasort directory */
-	char				*dir;
+	/* Datasort directory - chunks are merged into single sorted blob here */
+	char				*datasort_dir;
+	/* Chunks directory - if not defined, then chunks are stored in datasort_dir */
+	char				*chunks_dir;
```

Comment:
That's a bad practice, it introduces errors.
You have to check every place where this `if` happens, and it is at least 5 places in your patch: error removal, plain removal, tmp dir creation, chunk directory creation, split initialization.

Even more, you use both flag and chunk directory pointer, while you can only use one variable.